src/misc1.c

```diff
@@ -720,6 +720,8 @@ get_mode(char_u *buf)
 	    buf[i++] = 'v';
 	else if (exmode_active == EXMODE_NORMAL)
 	    buf[i++] = 'e';
+	if (cmdline_overstrike() == TRUE)
```

Comment:
When executing command in Ex mode (inside `do_cmdline()` but outside `getexline()`), Cmdline mode is not active and the result of `cmdline_overstrike()` is not valid, so a check for Cmdline mode is needed here.
```suggestion
	if ((State & MODE_CMDLINE) && cmdline_overstrike())
```

Reply:
i assumed ccline is zero initalized.
sorry am not used to going into new c code bases.